src/app/CommandHandler.h

```diff
@@ -170,13 +178,38 @@ class CommandHandler : public Messaging::ExchangeDelegate
      */
     void OnInvokeCommandRequest(Messaging::ExchangeContext * ec, const PayloadHeader & payloadHeader,
                                 System::PacketBufferHandle && payload, bool isTimedInvoke);
-    CHIP_ERROR AddStatus(const ConcreteCommandPath & aCommandPath, const Protocols::InteractionModel::Status aStatus);
 
-    // Same as AddStatus, but logs that the command represented by aCommandPath failed with the given
-    // error status and error message, if aStatus is an error. Errors on AddStatus are just logged
-    // (since the caller likely can only log and not further add a status).
-    void AddStatusAndLogIfFailure(const ConcreteCommandPath & aCommandPath, const Protocols::InteractionModel::Status aStatus,
-                                  const char * aMessage);
+    /**
+     * Adds the given command status and returns any failures in adding statuses (e.g. out
+     * of buffer space) to the caller
+     *
+     * [logging_option] controls what statuses are being logged.
+     */
+    CHIP_ERROR FailableAddStatus(const CommandPathStatus & status, const char * context = nullptr,
+                                 LogOption logging_option = LogOption::kFailures);
+
+    /**
+     * Adds a status when the caller is unable to handle any failures. Logging is performed
+     * and failure to register the status is checked with VerifyOrDie.
+     */
+    inline void AddStatus(const CommandPathStatus & status, const char * context = nullptr,
+                          LogOption logging_option = LogOption::kFailures)
+    {
+        VerifyOrDie(FailableAddStatus(status, context, logging_option) == CHIP_NO_ERROR);
+    }
+
+    inline CHIP_ERROR FailableAddStatus(const ConcreteCommandPath & aCommandPath, const Protocols::InteractionModel::Status aStatus,
+                                        const char * context = nullptr, LogOption logging_option = LogOption::kFailures)
+    {
+        return FailableAddStatus(CommandPathStatus(aCommandPath, aStatus), context, logging_option);
```

Comment:
So... what is the benefit of the CommandPathStatus here, over just passing through the path and status?  

It seems like extra complexity and a bit of extra overhead (to copy various bits around), but for not much reason....  especially if we remove the support for "log all" and "log none".